### PR TITLE
Added schema validation

### PR DESCRIPTION
**Pull request by sfc-gh-astus**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added or updated automated unit tests to verify correctness of my new code.
   * [ ] I've added or updated integration tests to verify correctness of my new code.
   * [ ] I've confirmed that my changes are working by executing CLI's commands manually.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [ ] I've described my changes in the release notes.
   * [ ] I've described my changes in the section below.

### Changes description
...

### CODE REVIEW

File: src/snowcli/cli/common/cli_global_context.py [hunk 1]
```diff
@@ -62,6 +64,12 @@ def schema(self) -> Optional[str]:
         return self._schema
 
     def set_schema(self, value: Optional[str]):
+        if (
+            value
+            and not re.compile("('|\").*('|\")$").match(value)
+            and re.compile(".+\..+").match(value)
```
Comment (sfc-gh-turbaszek) [inline src/snowcli/cli/common/cli_global_context.py:70]: Can you add comments for those patterns?

File: tests/test_common_global_context.py [hunk 2]
```diff
@@ -69,3 +71,17 @@ def test_connection_caching(mock_connect):
             ),
         ]
     )
+
+
+@pytest.mark.parametrize(
+    "schema", ["my_schema", "'.my_schema2'", '".my_schema3"', "'my.schema'"]
```
Comment (sfc-gh-turbaszek) [inline tests/test_common_global_context.py:77]:
```suggestion
    "schema", ["my_schema", '".my_schema2"', '".my_schema3"', '"my.schema"']
```

Comment (sfc-gh-turbaszek) [reply tests/test_common_global_context.py:77]: Quoted identifiers has to use `"`

Comment (sfc-gh-astus) [reply tests/test_common_global_context.py]: Ok

File: src/snowcli/cli/common/cli_global_context.py (hunk 1 above)
Comment (sfc-gh-astus) [reply src/snowcli/cli/common/cli_global_context.py]: First is replaced and second has comment

**Review by sfc-gh-turbaszek — APPROVED** [review]

**Pull request merged**